File: csharp/Hole2.Tests/TakeHomeCalculatorTest.cs
using Xunit;

namespace Hole2.Tests
{
    public class TakeHomeCalculatorTest
    {
        [Fact]
        public void CanCalculateTax()
        {
            int first = new TakeHomeCalculator(10).NetAmount(new Pair(40, "GBP"),
                new Pair(50, "GBP"),
                new Pair(60, "GBP")).value;
            Assert.Equal(135, first);
        }

        [Fact]
        public void CannotSumDifferentCurrencies()
        {
            Assert.Throws<Incalculable>(() => new TakeHomeCalculator(10).NetAmount(new
                Pair(4, "GBP"), new Pair(5, "USD")));
        }
    }
}

File: java/src/test/java/hole2/TakeHomeCalculatorTest.java
package hole2;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class TakeHomeCalculatorTest {

    @Test
    public void canCalculateTax() throws Exception {
        Integer first =
                new TakeHomeCalculator(10)
                        .netAmount(
                                new TakeHomeCalculator.Pair<>(40, "GBP"),
                                new TakeHomeCalculator.Pair<>(50, "GBP"),
                                new TakeHomeCalculator.Pair<>(60, "GBP"))
                        .first;
        assertThat(first.intValue()).isEqualTo(135);
    }

    @Test
    public void cannotSumDifferentCurrencies() throws Exception {
        assertThrows(
                Incalculable.class,
                () -> {
                    new TakeHomeCalculator(10)
                            .netAmount(
                                    new TakeHomeCalculator.Pair<>(4, "GBP"), new TakeHomeCalculator.Pair<>(5, "USD"));
                });
    }
}
